scripts/build/setup/termux_setup_gnu_as.sh

```diff
@@ -0,0 +1,53 @@
+# shellcheck shell=bash
+termux_setup_gnu_as() {
+	if [ "$TERMUX_ON_DEVICE_BUILD" = true ]; then
+		if [ -z "$(command -v as)" ]; then
+			echo "Command 'as' is not installed."
+			if [ "$TERMUX_APP_PACKAGE_MANAGER" = apt ]; then
+				echo "You can install it with"
+				echo
+				echo "  pkg install binutils"
+				echo
+			elif [ "$TERMUX_APP_PACKAGE_MANAGER" = pacman ]; then
+				echo "You can install it with"
+				echo
+				echo "  pacman -S binutils"
+				echo
+			fi
+			exit 1
+		fi
+		return
+	fi
+	if [ "$(basename "$NDK")" != android-ndk-r23c ]; then
+		if [ -n "$(command -v arm-linux-androideabi-as)" ] && [ -n "$(command -v aarch64-linux-android-as)" ] && \
+			[ -n "$(command -v i686-linux-android-as)" ] && [ -n "$(command -v x86_64-linux-android-as)" ]; then
+			return
+		fi
+		local NDK="$(dirname "$NDK")"/android-ndk-r23c
+		local CURRENT_TERMUX_NDK_VERSION_NUM="$TERMUX_NDK_VERSION_NUM"
+		local CURRENT_TERMUX_NDK_REVISION="$TERMUX_NDK_REVISION"
+		local TERMUX_NDK_VERSION_NUM=23
+		local TERMUX_NDK_REVISION="c"
+		sed -i "$TERMUX_SCRIPTDIR"/scripts/properties.sh \
+			-e "s|^TERMUX_NDK_VERSION_NUM=.*|TERMUX_NDK_VERSION_NUM=$TERMUX_NDK_VERSION_NUM|" \
+			-e "s|^TERMUX_NDK_REVISION=.*|TERMUX_NDK_REVISION=\"$TERMUX_NDK_REVISION\"|"
+		bash -c "cd $TERMUX_SCRIPTDIR; NDK=$NDK ./scripts/setup-android-sdk.sh"
+		sed -i "$TERMUX_SCRIPTDIR/scripts/properties.sh" \
+			-e "s|^TERMUX_NDK_VERSION_NUM=.*|TERMUX_NDK_VERSION_NUM=$CURRENT_TERMUX_NDK_VERSION_NUM|" \
+			-e "s|^TERMUX_NDK_REVISION=.*|TERMUX_NDK_REVISION=\"$CURRENT_TERMUX_NDK_REVISION\"|"
+		# https://github.com/android/ndk/issues/1569
+		# https://android-review.googlesource.com/c/platform/ndk/+/1817218
+		# these must be present for clang -fno-integrated-as to work correctly
+		# 1. bin/*-linux-android*-as
+		# 2. *-linux-android*/bin/as (symlink to #1)
+		# 3. lib/gcc/*-linux-android*/4.9.x/crtbegin.o (dummy file)
+		cp -f "$NDK"/toolchains/llvm/prebuilt/linux-x86_64/bin/{arm-linux-androideabi,{aarch64,i686,x86_64}-linux-android}-as \
```

Comment:
I think it would be better to add this stuff to a separate folder (maybe something like `~/.termux-build/_cache/android-r23c-gas-api-24-v0`) to make it similar to what we do for the ndk toolchain.